cmd/insprd/memory/tree/dapp_utils.go

```diff
@@ -302,3 +305,18 @@ func (amm *AppMemoryManager) recursiveBoundaryValidation(app *meta.App) error {
 
 	return nil
 }
+
+// SelectBrokerFromPriorityList takes a broker priority list and returns the first
+// broker that is available
+func SelectBrokerFromPriorityList(brokerList []string) string {
+	bmm := brokers.GetBrokerMemory()
+	availableBrokers := bmm.GetAll()
+
+	for _, broker := range brokerList {
+		if utils.Includes(availableBrokers, broker) {
+			return broker
+		}
+	}
+
+	return string(bmm.GetDefault())
+}
```

Comment:
I think it's better to wait for PR #61 to be merged, as it changes how the `GetDefault` works and this will affect this method